### DISCUSSION TITLE
duplicated class in product_listing Buy Now button for zen_image_button

### DISCUSSION THREAD

**Discussion by torvista** [zencart/zencart#5827] [Ideas]
Multiple HTML Warnings
: <span> dropping value "cssButton normal_button button button_buy_now" for repeated attribute "class".

on each Buy Now button.

In
includes/modules/responsive_classic/product_listing.php
`$lc_button = '<div class="cart-add"><a href="' . zen_href_link($_GET['main_page'], zen_get_all_get_params(array('action')) . 'action=buy_now&products_id=' . $listing->fields['products_id']) . '">' . zen_image_button(BUTTON_IMAGE_BUY_NOW, BUTTON_BUY_NOW_ALT, 'class="listingBuyNowButton"') . '</a></div>';`

if
`zen_image_button(BUTTON_IMAGE_BUY_NOW, BUTTON_BUY_NOW_ALT, 'class="listingBuyNowButton"`
is changed to
`zen_image_button(BUTTON_IMAGE_BUY_NOW, BUTTON_BUY_NOW_ALT, '', 'listingBuyNowButton`
this clears the errors as this fourth variable gets correctly used as the secondary class in zen_image_button and zenCssButton insted of being duplicated via the paramter variable.

However the handling of the third parameter variable in zen_image_button conflicts with this, as another class may be added for a rollover. I think the parameter variable must be strictly not a class.
So I think this function needs review, with subsequent review of all uses where the third parameter is passed, to ensure it is not a class.

**Comment by drbyte** [comment]
Ya I’d like to get rid of all that old rollover class stuff. It’s outdated.

**Comment by torvista** [comment]
>get rid of all that old rollover class stuff
> I think this function needs review, with subsequent review of all uses where the third parameter is passed, to ensure it is not a class.

I'm not likely to "fix" this, as "fix" is undefined / I can see a rabbit-hole beckoning that I am not in a position to jump into.
